### ISSUE TITLE
Rendering error when reloading an external .html in a div

### ISSUE THREAD

**Issue by JeanLouisMassei** [vitmalina/w2ui#1843]
Hello :)

Thanks, first, for the great job you made, w2ui is incredible !! :D:D

So I am having trouble reloading external .html in div. In my example I have a top button which (re)load grid.html on that div. But it seems that the scripts in the end of grid.html is not rendered in the div, if I click again on the Load button..

Am i doing something wrong ?

Here are the files : 
https://github.com/JeanLouisMassei/w2ui-bug

Edit : seems there is a timeout bug somwhere..

$("#layout").load("grid.html"); 
//setTimeout(function(){ $("#layout").load("grid.html"); }, 2000); // appears then disappear
setTimeout(function(){ $("#layout").load("grid.html"); }, 50); // just appears, ok

**Comment by JeanLouisMassei** [comment]
Ok, found the solution :

main.php
`<!DOCTYPE html>
<html>
<head>
    <title>W2UI Demo: combo-1</title>
    <script src="http://ajax.googleapis.com/ajax/libs/jquery/2.1.0/jquery.min.js"></script>
    <script type="text/javascript" src="http://rawgit.com/vitmalina/w2ui/master/dist/w2ui.min.js"></script>
    <link rel="stylesheet" type="text/css" href="http://rawgit.com/vitmalina/w2ui/master/dist/w2ui.min.css" />
</head>
<body>

<div id="main" style="position: absolute; top: -1px; left: 0px; bottom: 0px; right: 0px;"></div>

<script type="text/javascript">

    $.getScript('config.js', function() {

        config.sidebar = {
            name: 'sidebar',
            nodes: [ 
                { 
                    id: 'general', text: 'General', group: true, expanded: true, nodes: [
                    { id: 'grid1', text: 'Grid1', img: 'icon-page' },
                    { id: 'form1', text: 'Form1', img: 'icon-page' },
                    { id: 'form2', text: 'Form2', img: 'icon-page' }
                ]}
            ],
            onClick: function (event) {
                w2ui.layout.content('main', w2ui[event.target]);
            }
        };

        $('#main').w2layout(config.layout);
        w2ui.layout.content('left', $().w2sidebar(config.sidebar));
        w2ui.layout.content('main', '');
        $(w2ui.layout.el('main')).css({ 'border-left': '1px solid silver'});

        $().w2form(config.form1);
        $().w2form(config.form2);
        $().w2grid(config.grid1);
    });

</script>

</body>
</html>`

config.js
`var config = {

    layout: {
        name: 'layout',
        padding: 0,
        panels: [
            { type: 'left', size: 200, resizable: true, minSize: 120 },
            { type: 'main', minSize: 550, overflow: 'hidden' }
        ]
    },

    grid1: {
    	name: 'grid1',
        columns: [
            { field: 'art', caption: 'Article', size: '15%', sortable: true, searchable: true },
            { field: 'mou', caption: 'Mouvement', size: '15%' },
            { field: 'sto', caption: 'Stock', size: '15%', sortable: true, searchable: true },
            { field: 'pro', caption: 'Projet', size: '15%', sortable: true, searchable: true },
            { field: 'doc', caption: 'Document', size: '15%', sortable: true, searchable: true },
            { field: 'dat', caption: 'Date', size: '100px', render: 'date:dd/mm/yyyy', sortable: true, searchable: true }
        ],
        records: [
            { recid: 1, art: 'B0014', mou: '-5', sto: 'CONTENEUR 1', pro: '326', doc: 'BDS 0012', dat: '04/03/2012' },
            { recid: 2, art: 'B0014', mou: '6', sto: 'CONTENEUR 1', pro: '326', doc: 'CF BBI BGI 2018 0452-5', dat: '03/15/2012' }
        ],
    },
    
    form1: {
    	name   : 'form1',
        header : 'Auto-Generated Form',
        url    : 'server/post',
        fields : [
            { field: 'first_name', type: 'text', required: true, html: { caption: 'First Name', attr: 'style="width: 300px"' } },
            { field: 'last_name',  type: 'text', required: true, html: { caption: 'Last Name', attr: 'style="width: 300px"' } },
            { field: 'comments',   type: 'textarea', html: { caption: 'Comments', attr: 'style="width: 300px; height: 90px"' } }
        ],
        actions: {
            'Save': function (event) {
                console.log('save', event);
                this.save();
            },
            'Clear': function (event) {
                console.log('clear', event);
                this.clear();
            }
        }
    },

    form2: {
    	name: 'form2',
	    formHTML: 	    		
	    	'<div class="w2ui-page page-0">'+
		    '    <div class="w2ui-field">'+
		    '        <label>First Name:</label>'+
		    '        <div>'+
		    '            <input name="first_name" type="text" maxlength="100" size="60"/>'+
		    '        </div>'+
		    '    </div>'+
		    '    <div class="w2ui-field">'+
		    '        <label>Last Name:</label>'+
		    '        <div>'+
		    '            <input name="last_name" type="text" maxlength="100" size="60"/>'+
		    '        </div>'+
		    '    </div>'+
		    '    <div class="w2ui-field">'+
		    '        <label>Comments:</label>'+
		    '        <div>'+
		    '            <textarea name="comments" type="text" style="width: 100%; height: 80px; resize: none"></textarea>'+
		    '        </div>'+
		    '    </div>'+
		    '</div>'+
		    '<div class="w2ui-page page-1">'+
		    '    <div class="w2ui-field">'+
		    '        <label>Address:</label>'+
		    '        <div>'+
		    '            <input name="address1" type="text" maxlength="100" style="width: 100%"/>'+
		    '        </div>'+
		    '    </div>'+
		    '    <div class="w2ui-field">'+
		    '        <label>Line 2:</label>'+
		    '        <div>'+
		    '            <input name="address2" type="text" maxlength="100" style="width: 100%"/>'+
		    '        </div>'+
		    '    </div>'+
		    '    <div class="w2ui-field">'+
		    '        <label>City:</label>'+
		    '        <div>'+
		    '            <input name="city" type="text" maxlength="50" size="25"/>'+
		    '        </div>'+
		    '    </div>'+
		    '    <div class="w2ui-field">'+
		    '        <label>State:</label>'+
		    '        <div>'+
		    '            <input name="state" type="text" maxlength="2" size="2"/>'+
		    '        </div>'+
		    '    </div>'+
		    '    <div class="w2ui-field">'+
		    '        <label>Zip:</label>'+
		    '        <div>'+
		    '            <input name="zip" type="text" maxlength="10" size="10"/>'+
		    '        </div>'+
		    '    </div>'+
		    '</div>'+
		    '<div class="w2ui-page page-2">'+
		    '    <div class="w2ui-field">'+
		    '        <label>Short Bio:</label>'+
		    '        <div>'+
		    '            <textarea name="short_bio" type="text" style="width: 100%; height: 80px; resize: none"></textarea>'+
		    '        </div>'+
		    '    </div>'+
		    '    <div class="w2ui-field">'+
		    '        <label>Talk Name:</label>'+
		    '        <div>'+
		    '            <input name="talk_name" type="text" maxlength="100" style="width: 100%"/>'+
		    '        </div>'+
		    '    </div>'+
		    '    <div class="w2ui-field">'+
		    '        <label>Talk Description:</label>'+
		    '        <div>'+
		    '           <textarea name="description" type="text" style="width: 100%; height: 80px; resize: none"></textarea>'+
		    '        </div>'+
		    '    </div>'+
		    '</div>'+
		    '<div class="w2ui-buttons">'+
		    '    <button class="w2ui-btn" name="reset">Reset</button>'+
		    '    <button class="w2ui-btn" name="save">Save</button>'+
		    '</div>',
		tabs: [
            { id: 'tab1', caption: 'General' },
            { id: 'tab2', caption: 'Address'},
            { id: 'tab3', caption: 'About' }
        ],
        fields : [
            { field: 'first_name', type: 'text', required: true },
            { field: 'last_name',  type: 'text', required: true },
            { field: 'comments',   type: 'text'},
            { field: 'address1', type: 'text', required: true },
            { field: 'address2', type: 'text' },
            { field: 'city', type: 'text', required: true },
            { field: 'state', type: 'text', required: true },
            { field: 'zip', type: 'int', required: true },
            { field: 'short_bio', type: 'text' },
            { field: 'talk_name', type: 'text', required: true },
            { field: 'description', type: 'text' }
        ],
        actions: {
            reset: function () {
                this.clear();
            },
            save: function () {
                this.save();
            }
        }
    }
};`

**Comment by vitmalina** [comment]
It is not a time out bug. The reason is likely because you hit Load button twice. If you try to create grid again, it will error off (see console) because object with this name already exists. So, yo must destroy it if it exists.

**Comment by JeanLouisMassei** [comment]
Indeed
https://stackoverflow.com/questions/33461626/w2ui-grid-re-rendering-issue
Thanks !!!

**Comment by JeanLouisMassei** [comment]
Hello :) 

I have spoken to quickly..
In this example (w2ui rc1 + jquery 2.1.1) sometime some pages don't load : 
![Capture d’écran 2019-07-04 à 20 16 42](https://user-images.githubusercontent.com/10390131/60683617-00ac7d80-9e99-11e9-8391-e3e9549ee5c8.png)
https://github.com/JeanLouisMassei/w2ui-bug

Can you tell me which versions of w2ui / jqery are fully stable ?

Thanks !
